File: codegen/results/v001/qwen3.5_27b-nvfp4_nothink/20260408_150810/python/004_json_filter/solution.py
import json

# Read the input data
try:
    with open('input/data.json', 'r') as f:
        data = json.load(f)
except FileNotFoundError:
    print("[]")
    exit()
except json.JSONDecodeError:
    print("[]")
    exit()

# Filter and sort the records
filtered_data = [
    record for record in data
    if isinstance(record, dict) 
    and record.get('active') is True 
    and isinstance(record.get('age'), int) 
    and record['age'] >= 30
]

sorted_data = sorted(filtered_data, key=lambda x: x.get('name', ''))

# Output the result to stdout
print(json.dumps(sorted_data))

File: codegen/results/v001/qwen3.5_27b-nvfp4_nothink/20260408_150810/typescript/004_json_filter/solution.ts
import * as fs from 'fs';
import * as path from 'path';

// Define the interface for the data objects
interface Person {
  name: string;
  age: number;
  active: boolean;
  score: number;
}

// Resolve the file path
const filePath = path.join('input', 'data.json');

try {
  // Read and parse the JSON file
  const rawData = fs.readFileSync(filePath, 'utf8');
  const data: Person[] = JSON.parse(rawData);

  // Filter records where active is true and age is >= 30
  const filtered = data.filter(person => person.active && person.age >= 30);

  // Sort the filtered records by name ascending
  filtered.sort((a, b) => a.name.localeCompare(b.name));

  // Output the result as JSON to stdout
  console.log(JSON.stringify(filtered));
} catch (error) {
  // If there's an error (e.g., file not found), exit with an error code
  process.exit(1);
}